src/chatty/Helper.java

```diff
@@ -72,6 +72,57 @@ public interface ParseChannelHelper {
      * @param prepend Whether to prepend # if necessary
      * @return Set of channels sorted as in the String
      */
+
+
+    private static HashMap<String,Boolean> setPossibleChannelType(String[] catSplit, HashMap<String,Boolean> channelType){
+
+        for (int i = 1; i < catSplit.length; i++) {
+            if (catSplit[i].equals("#")) {
+                channelType.put("onlyChans",true);
+            }
+            else if (catSplit[i].equals("!#")) {
+                channelType.put("noChans",true);
+
+            }
+            else if (catSplit[i].equals("live")) {
+                channelType.put("onlyLive",true);
+            }
+        }
+        return channelType;
+    }
+
+    private static List<String> storePossibleChannelNamesToList(String cat, HashMap<String,Boolean> channelType){
+        List<String> chans = new ArrayList<>();
+        if (cat.equals("*")) {
+            chans = new ArrayList<>(parseChannelHelper.getFavorites());
+        }
+        else {
+            for (String name : parseChannelHelper.getNamesByCategory(cat)) {
+                if ((!channelType.get("noChans") || !name.startsWith("#"))
+                        && (!channelType.get("onlyChans") || name.startsWith("#"))) {
+                    chans.add(name);
+                }
+            }
+        }
+        return chans;
+    }
+
+    private static void getPossibleChannelNames(String channel, boolean prepend, Set<String> result){
+        String[] channelTypes = new String[]{"noChans","onlyChans","onlyLive"};
+        HashMap<String,Boolean> channelTypeHM = new HashMap<>();
+        for(String channelType: channelTypes){
```

Comment:
1. In existing code `for` and its `(` are separated by a space.
2. In existing code that you’re replacing the `Type variable : class` uses ` : `.
3. `){` as mentioned elsewhere.

```suggestion
        for (String channelType : channelTypes) {
```